### PR TITLE
[🐛 🔨 ] set_action_for_agent expects a ActionTuple with batch size 1.

### PR DESCRIPTION
**Pull request by vincentpierre**
### Proposed change(s)

Fixed validate_action to expect the right dimensions when set_action_single_agent is called.
Added a test.

### Types of change(s)

- [x] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [x] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/main/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/main/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/main/docs/Migrating.md) (if applicable)

### CODE REVIEW

File: ml-agents-envs/mlagents_envs/environment.py [hunk 1]
```diff
@@ -366,7 +366,7 @@ def set_action_for_agent(
             return
         action_spec = self._env_specs[behavior_name].action_spec
         num_agents = len(self._env_state[behavior_name][0])
```
Comment (andrewcoh) [inline ml-agents-envs/mlagents_envs/environment.py:368]: this line can be killed right?

File: ml-agents-envs/mlagents_envs/environment.py [hunk 2]
```diff
@@ -366,7 +366,7 @@ def set_action_for_agent(
...
-        action = action_spec._validate_action(action, None, behavior_name)
+        action = action_spec._validate_action(action, 1, behavior_name)
         if behavior_name not in self._env_actions:
             self._env_actions[behavior_name] = action_spec.empty_action(num_agents)
```
Comment (andrewcoh) [inline ml-agents-envs/mlagents_envs/environment.py:371]: can we set this to 1?

**Review by andrewcoh — APPROVED** [review]

File: ml-agents-envs/mlagents_envs/environment.py (hunk 2 above)
Comment (vincentpierre) [reply ml-agents-envs/mlagents_envs/environment.py:371]: What do you mean?/ You mean by default?

Comment (andrewcoh) [reply ml-agents-envs/mlagents_envs/environment.py:371]: oh I remember, all other actions are zero'd except the action being set for the agent.

**Pull request merged**